hnn_core/network.py

```diff
@@ -1196,12 +1198,81 @@ def _add_cell_type(self, cell_name, pos, cell_template=None):
         ll = self._n_gids
         self._n_gids += len(pos)
         self.gid_ranges[cell_name] = range(ll, self._n_gids)
-
         self.pos_dict[cell_name] = pos
         if cell_template is not None:
             self.cell_types.update({cell_name: cell_template})
             self._n_cells += len(pos)
 
+    def rename_cell_type(self, original_name, new_name):
+        """Renames cell types in the network.
+
+        Parameters
+        ----------
+        original_name: str
+            The original cell name in the network to be changed.
+        new_name: str
+            The desired new cell name in the network.
+        """
+        _validate_type(original_name, str, 'original_name')
+        _validate_type(new_name, str, 'new_name')
+        if original_name not in self.cell_types.keys():
+            # Raises error if the original name is not in cell_types
+            raise ValueError(f"'{original_name}' is not in cell_types!")
+        elif new_name in self.cell_types.keys():
+            # Raises error if the new name is already in cell_types
+            raise ValueError(f"'{new_name}' is already in cell_types!")
+        elif original_name in self.cell_types.keys():
+            # Update cell name in dicts/etc. where order doesn't matter
+            # Update Network.cell_types
+            self.cell_types[new_name] = self.cell_types.pop(original_name)
+            # `Cell.name` does not currently provide a way to change its value,
+            # so we will leave that untouched for now.
+            # Update Network.pos_dict
+            self.pos_dict[new_name] = self.pos_dict.pop(original_name)
+            # Update Network.external_biases
+            for bias_key, bias_value in self.external_biases.items():
+                if original_name in self.external_biases[bias_key].keys():
+                    self.external_biases[bias_key][new_name] = \
+                        self.external_biases[bias_key].pop(original_name)
+            # Update Network.external_drives
+            for drive_key, drive_config in self.external_drives.items():
+                if original_name in drive_config['target_types']:
+                    drive_config['target_types'].remove(original_name)
+                    drive_config['target_types'].append(new_name)
+                    drive_config['target_types'].sort()
+                    for config_key, config_value in drive_config.items():
+                        if (config_key == 'dynamics'):
```

Comment:
```suggestion
                        if config_key == 'dynamics':
```

just a stylistic suggestion